crates/app/src/handle.rs

```diff
@@ -55,7 +55,7 @@ pub async fn handle(mut req: Request<Body>) -> impl IntoResponse {
     })?;
     assert_eq!(extensions.insert(repo), None, "duplicate repository name");
 
-    let mut tail = tail.split_terminator('/');
+    let mut tail = tail.splitn(4, '/');
```

Comment:
Isn't this off-by-one?
E.g. for `rail` equal to `_tag/v0.1.2/tree/a/b/c`, `path` on `L97` should contain `a/b/c`

Reply:
> If `n` substrings are returned, the last substring (the `n`th substring) will contain the remainder of the string.

So this is good https://doc.rust-lang.org/std/primitive.str.html#method.splitn